build.zig

```
@@ -42,18 +36,20 @@ pub fn build(b: *std.Build) void {
     const clap = b.dependency("clap", .{ .target = target, .optimize = optimize });
     exe.root_module.addImport("clap", clap.module("clap"));
 
+    exe.addIncludePath(.{ .path = "include" });
     exe.linkSystemLibrary("pam");
     exe.linkSystemLibrary("xcb");
     exe.linkLibC();
 
-    exe.addIncludePath(.{ .path = "dep/termbox_next/src" });
-
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/input.c" }, .flags = &c_args });
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/memstream.c" }, .flags = &c_args });
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/ringbuffer.c" }, .flags = &c_args });
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/term.c" }, .flags = &c_args });
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/termbox.c" }, .flags = &c_args });
-    exe.addCSourceFile(.{ .file = .{ .path = "dep/termbox_next/src/utf8.c" }, .flags = &c_args });
+    // Only fails with ReleaseSafe, so we'll override it.
+    const translate_c = b.addTranslateC(.{
+        .root_source_file = .{ .path = "include/termbox2.h" },
+        .target = target,
+        .optimize = if (optimize == .ReleaseSafe) .ReleaseFast else optimize,
```

We definitely shouldn't do that. Since this is most likely a Zig compiler bug, let it crash with a runtime error for now if ReleaseSafe is used. (Perhaps it has been fixed in Zig 0.13.0?)

---

I'm not sure completely breaking builds in ReleaseSafe is desired when only a single step needs to be overridden, especially when it is considered the "main release mode". 

Editing the comment above it to read: `// HACK: Only fails with ReleaseSafe, so we'll override it for now`, would be a better indicator to change it in the future.

---

I'm more concerned about the safety issues it may create. In ReleaseSafe, safety checks (like overflowing, array index out of bounds, etc...) are kept, but not in ReleaseFast. This could potentially dismiss some bugs in Ly by making it automatically build in ReleaseFast even when ReleaseSafe is selected.

---

As I understand it:
The rest of the project should still have safety checks outside of that module.
Without it, you're forced to compile the whole project with ReleaseFast/ReleaseSmall, so there would be no safety checks at all. 

This temporary workaround is the best case scenario imo.

---

Fair enough. However, we should not just have a comment indicating the hack, but also print a message to potentially warn the user (this may not matter for automated buildbots like for distributions, but it's still better than othing).

---

I added a warning.